src/command/backup/backup.c

```diff
@@ -850,6 +850,17 @@ backupStart(BackupData *backupData)
             DbBackupStartResult dbBackupStartResult = dbBackupStart(
                 backupData->dbPrimary, cfgOptionBool(cfgOptStartFast), cfgOptionBool(cfgOptStopAuto));
 
+            // When the start-fast option is disabled and db-timeout is smaller than checkpoint_timeout, the command may timeout
+            // before the backup actually starts
+            if (!cfgOptionBool(cfgOptStartFast) &&
+                cfgOptionInt64(cfgOptDbTimeout) <= (int64_t)(dbCheckpointTimeout(backupData->dbPrimary) * MSEC_PER_SEC))
+            {
+                LOG_WARN_FMT(
+                    CFGOPT_START_FAST " is disabled and " CFGOPT_DB_TIMEOUT " (%ss) is smaller than the checkpoint_timeout (%us)"
+                    " reported by the database - timeout may occur before the backup actually starts",
+                    strZ(cfgOptionDisplay(cfgOptDbTimeout)), dbCheckpointTimeout(backupData->dbPrimary));
+            }
+
```

Comment:
Hm I see, `db-timeout` might be different on repo/db hosts. I'll investigate how to handle the info from `dbOpenProtocol` then :-)

Reply:
The "not remote" part seems to be working now (since 80358f3) but I'm still struggling with `dbOpenProtocol()`.

Should I change the `pckWriteU32P` to push both index and timeout or should I rather change the function return with something like `FUNCTION_LOG_RETURN(UINT64, cfgOptionUInt64(cfgOptDbTimeout));`?

Reply:
I remembered that we already have an API for pulling options from a remote host, so I added that in 31e71c80d2f1. 

Also, I changed `pgClientTimeout()` to `cfgOptionUInt64(cfgOptDbTimeout)` for the local case. The client timeout is protocol-timeout which is >= db-timeout + 30.

Now we'll see if this passes CI.

Reply:
Oh, that's great, thanks :-)